src/Compilers/VisualBasic/Test/Emit/Emit/EditAndContinue/EditAndContinueStateMachineTests.vb

```diff
@@ -8217,6 +8217,227 @@ End Class")
+
+                Dim reader0 = md0.MetadataReader
+                Dim generation0 = EmitBaseline.CreateInitialBaseline(md0, AddressOf v0.CreateSymReader().GetEncMethodDebugInfo)
+
+                ' Notice encLocalSlotMap CDI on both M and MoveNext methods.
+                ' The former is used to calculate mapping for variables lifted to fields of the state machine,
+                ' the latter is used to map local variable slots in the MoveNext method.
+                ' Here, the variable lifted to the state machine field is the closure pointer storage.
+                v0.VerifyPdb("
+<symbols>
+  <methods>
+    <method containingType=""C"" name=""M"">
+      <customDebugInfo>
+        <forwardIterator name=""VB$StateMachine_1_M"" />
+        <encLocalSlotMap>
+          <slot kind=""30"" offset=""-1"" />
+        </encLocalSlotMap>
+        <encLambdaMap>
+          <methodOrdinal>1</methodOrdinal>
+          <closure offset=""-1"" />
+          <lambda offset=""142"" closure=""0"" />
+        </encLambdaMap>
+        <encStateMachineStateMap>
+          <state number=""0"" offset=""74"" />
+        </encStateMachineStateMap>
+      </customDebugInfo>
+    </method>
+    <method containingType=""C+VB$StateMachine_1_M"" name=""MoveNext"">
+      <customDebugInfo>
+        <hoistedLocalScopes>
+          <slot startOffset=""0x0"" endOffset=""0xe3"" />
+        </hoistedLocalScopes>
+        <encLocalSlotMap>
+          <slot kind=""27"" offset=""-1"" />
+          <slot kind=""33"" offset=""74"" />
+          <slot kind=""temp"" />
+          <slot kind=""temp"" />
+        </encLocalSlotMap>
+      </customDebugInfo>
+      <asyncInfo>
+        <kickoffMethod declaringType=""C"" methodName=""M"" />
+        <await yield=""0x44"" resume=""0x62"" declaringType=""C+VB$StateMachine_1_M"" methodName=""MoveNext"" />
+      </asyncInfo>
+    </method>
+    <method containingType=""C+_Closure$__1-0"" name=""_Lambda$__0"">
+      <customDebugInfo>
+        <encLocalSlotMap>
+          <slot kind=""21"" offset=""142"" />
+        </encLocalSlotMap>
+      </customDebugInfo>
+    </method>
+  </methods>
+</symbols>
+", options:=PdbValidationOptions.ExcludeDocuments Or PdbValidationOptions.ExcludeSequencePoints Or PdbValidationOptions.ExcludeNamespaces Or PdbValidationOptions.ExcludeScopes,
+   format:=DebugInformationFormat.PortablePdb)
+
+                CheckNames(reader0, reader0.GetTypeDefNames(), "<Module>", "C", "VB$StateMachine_1_M", "_Closure$__1-0")
+
+                Dim diff1 = compilation1.EmitDifference(
+                generation0,
+                ImmutableArray.Create(
+                    SemanticEdit.Create(SemanticEditKind.Update, m0, m1, GetSyntaxMapFromMarkers(source0, source1), preserveLocalVariables:=True)))
+
+                ' Notice that we reused field "$VB$ResumableLocal_$VB$Closure_$0" (there is no "$VB$ResumableLocal_$VB$Closure_$1"), which stores the local function closure pointer.
```

Comment:
>local function closure [](http://example.com/codeflow?start=150&length=22)

Minor: Should this be simply "closure" since there are no local functions?